docs: Improve generated docs formatting

dnaka91 opened the pull request:
<!--- Please fill out the following template, which will help other contributors review your Pull Request. -->

<!--- Make sure you’ve read the contribution guidelines here: https://github.com/obsproject/obs-websocket/wiki/Contributing-Guidelines -->

### Description
<!--- Describe your changes. -->
Update the docs generation to make the docs markdown (and all other README.md as well actually) pass the default settings of markdownlint. With a few tiny exceptions here and there.

The **line length** rule is disabled globally as most editors do automatic line wrapping for us anyways.


### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes/closes an open issue or implements feature request, -->
<!--- please link to the issue here. -->
[markdowlint](https://github.com/DavidAnson/markdownlint) is a nice tool to check for common best practices that keep the Markdown clean and readable. Most of the code is generated and it probably _doesn't really matter_, but still it feels better to have a nice and clean output.



### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes, along with the OS(s) you tested with. -->
Tested OS(s): 

### Types of changes
<!--- What types of changes does your PR introduce? Uncomment all that apply -->

<!--- - Bug fix (non-breaking change which fixes an issue) -->
<!--- - Performance enhancement (non-breaking change which improves efficiency) -->
<!--- - Code cleanup (non-breaking change which makes code smaller or more readable) -->
<!--- - New request/event (non-breaking) -->
- Documentation change (a change to documentation pages)
<!--- - Other Enhancement (anything not applicable to what is listed) -->

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
-  [x] I have read the [Contributing Guidelines](https://github.com/obsproject/obs-websocket/wiki/Contributing-Guidelines).
-  [x] All commit messages are properly formatted and commits squashed where appropriate.
-  [x] My code is not on the `master` branch.
-  [x] The code has been tested.
-  [x] I have included updates to all appropriate documentation.

tt2468 commented on docs/docs/generate_md.py:
```diff
@@ -160,7 +159,7 @@ def get_requests(requests):
             continue
         categoryFragment = get_fragment(category)
-        ret += '\n\n## {}\n\n'.format(category)
+        ret += '## {} requests\n\n'.format(category)
```
Should `requests` be capitalized?

tt2468 commented on src/requesthandler/RequestHandler_Inputs.cpp:
```diff
@@ -549,6 +549,7 @@ RequestResult RequestHandler::SetInputAudioSyncOffset(const Request& request)
  * Gets the audio monitor type of an input.
  *
  * The available audio monitor types are:
+ *
```
Can you go over the rest of the documentation and make sure it's formatted correctly?

dnaka91 replied:
Sorry, will finally get back to this now.
Probably a few updates needed here and there but should be quick overall.

dnaka91 commented:
Just pulled in the latest changes from master and adjusted remaining issues. All markdown files in the project look good now.

As a plus, I added a new workflow to the project that updates docs and runs `markdownlint-cli` on all Markdown files, so we directly see in the CI if some formatting is off.
